File: leetcode/easy/running-sum-of-1d-array.cpp
// https://leetcode.com/problems/running-sum-of-1d-array/
class Solution {
public:
    vector<int> runningSum(vector<int>& nums) 
    {
       for (int i = 1; i < nums.size(); ++i) 
       {
            nums[i] += nums[i - 1];
        }
        return nums;
    }
};


class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        vector<int> res(nums.size());
        int count = 0;
        for(int i = 0; i < nums.size(); i++){
            count += nums[i];
            res[i] = count;
        }
        return res;
    }
};

File: array/running_sum_of_1d_array.go
package array

func runningSum(nums []int) []int {
	var res = make([]int, len(nums))
	for i, val := range nums {
		if i == 0 {
			res[0] = val
			continue
		}
		res[i] = val + res[i-1]
	}
	return res
}
